astropy/io/misc/hdf5.py

```diff
@@ -247,12 +263,16 @@ def write_table_hdf5(table, output, path=None, compression=False,
     else:
         dset = output_group.create_dataset(name, data=table.as_array())
 
-    # Write the meta-data to the file
-    for key in table.meta:
-        val = table.meta[key]
-        try:
-            dset.attrs[key] = val
-        except TypeError:
-            warnings.warn("Attribute `{0}` of type {1} cannot be written to "
-                          "HDF5 files - skipping".format(key, type(val)),
-                          AstropyUserWarning)
+    if serialize_meta:
+        header_yaml = meta.get_yaml_from_table(table)
+        dset.attrs[META_KEY] = [h.encode('utf8') for h in header_yaml]
+    else:
+        # Write the meta-data to the file
+        for key in table.meta:
+            val = table.meta[key]
+            try:
+                dset.attrs[key] = val
+            except TypeError:
+                warnings.warn("Attribute `{0}` of type {1} cannot be written to "
+                              "HDF5 files - skipping".format(key, type(val)),
```

Comment:
After `skipping` add `(consider specifying serialize_meta=True to write all meta data)`.